Code/GraphMol/RGroupDecomposition/testRGroupDecomp.cpp

```diff
@@ -2690,6 +2690,78 @@ M  END
   CHECK_RGROUP(it, expected);
 }
 
+void testDoNotChooseUnrelatedCores() {
+  BOOST_LOG(rdInfoLog)
+      << "********************************************************\n";
+  BOOST_LOG(rdInfoLog) << "Test that later cores with more R-groups\n"
+                       << "are only chosen if superstructures of earlier\n"
+                       << "cores" << std::endl;
+  {
+    // 1st test, two unrelated cores:
+    // 1) 5 terms, 1 R-group
+    // 2) 6 terms, 3 R-groups
+    // dataset molecule can fit core 1 adding 1 non-user defined R label,
+    // and core 2 with no addition of R labels
+    ROMOL_SPTR core5Terms1RGroup = "[*:1]C1COCN1"_smiles;
```

Comment:
being able to initialize shared pointers from unique pointers like this is cool. I always forget that this is possible.